pkg/cmd/admin/registry/registry.go

```diff
@@ -207,30 +205,44 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
 		if len(remove) > 0 {
 			return kcmdutil.UsageError(cmd, "You may not pass negative labels in selector %q", cfg.Selector)
 		}
-		nodeSelector = valid
+		cfg.nodeSelector = valid
 	}
 
-	image := cfg.ImageTemplate.ExpandOrDie(cfg.Type)
+	if ports, err := app.ContainerPortsFromString(cfg.Ports); err != nil {
+		return err
+	} else {
+		cfg.ports = ports
+	}
 
-	namespace, _, err := f.OpenShiftClientConfig.Namespace()
-	if err != nil {
-		return fmt.Errorf("error getting client: %v", err)
+	if namespace, _, err := f.OpenShiftClientConfig.Namespace(); err != nil {
+		return fmt.Errorf("error getting namespace: %v", err)
+	} else {
+		cfg.namespace = namespace
 	}
-	_, kClient, err := f.Clients()
-	if err != nil {
+
+	if _, kClient, err := f.Clients(); err != nil {
 		return fmt.Errorf("error getting client: %v", err)
+	} else {
+		cfg.kClient = kClient
 	}
 
 	cfg.Action.Bulk.Mapper = clientcmd.ResourceMapper(f)
 	cfg.Action.Out, cfg.Action.ErrOut = out, cmd.Out()
 	cfg.Action.Bulk.Op = configcmd.Create
 
+	return nil
+}
+
+// RunCmdRegistry contains all the necessary functionality for the OpenShift cli registry command
+func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *RegistryConfig, args []string) error {
```

Comment:
Also, usually we never bring stuff like cobra or the factory down to this level. Every single bit you need should be a part of the config. Actually I think your config should be different: you shoulf have a top-level RegistryOptions that would include RegistryConfig and every other machinery. Then you should have Complete, Validate, and Run on RegistryOptions.